kernel/kernel-api/src/main/java/io/delta/kernel/internal/commitrange/CommitRangeFactory.java

```diff
@@ -94,7 +101,8 @@ private long resolveStartVersion(Engine engine) {
    * timestamp to version. When the boundary is not specified, this returns empty, as the version
    * cannot be resolved until later after we have performed any listing.
    */
-  private Optional<Long> resolveEndVersionIfSpecified(Engine engine) {
+  private Optional<Long> resolveEndVersionIfSpecified(
+      Engine engine, List<ParsedCatalogCommitData> parsedDeltas) {
```

Comment:
`parsedCatalogCommits`

if this was `List<ParsedDeltaData>` (the parent) then we should call it `parsedDeltas`